torch/_inductor/codegen/wrapper.py

```diff
@@ -1062,6 +1072,32 @@ def codegen_model_constructor(self):
                         f"inputs_info_[{idx}].shape.emplace_back({size}, {size}, nullptr);"
                     )
 
+            constant_offset_mapping = dict(V.graph.aot_constant_offset)  # noqa: C416
+            for idx, (name, tensor) in enumerate(V.graph.constants.items()):
+                assert isinstance(tensor, torch.Tensor)
+                self.prefix.writeline(f"""constants_info_[{idx}].name = "{name}";""")
+                self.prefix.writeline(
+                    f"constants_info_[{idx}].dtype = {DTYPE_TO_ATEN[tensor.dtype]};"
+                )
+                self.prefix.writeline(
+                    f"constants_info_[{idx}].offset = {constant_offset_mapping[name]};"
+                )
+                tensor = tensor.contiguous()
```

Comment:
We should not be making tensors contiguous, this is going to silently give wrong with results with e.g. channels last constants. 

You can just serialize the raw storage, `tensor.untyped_storage()`, and then use the size, stride, storage_offset with the storage